### ISSUE TITLE
Unexpected To/From selection if sender's From field matches a stored identity

### ISSUE THREAD

**Issue by mjnewton** [roundcube/roundcubemail#8603]
Hopefully I can describe this issue better than what I've achieved so far with the title...

Roundcube v1.4.11

Scenario:

1. I am Bob. My email address is bob@mydomain.com
2. I have an additional identity (not set to default) of alice@mydomain.com so that I can send email on behalf of Alice if need be
3. Alice sends an email from alice@mydomain.com to me at bob@mydomain.com

Issue:

4. I hit reply and the From: field gets populated with alice@mydomain.com and the To: field with bob@mydomain.com
5. If I don't spot this (and manually swap the fields around) the message gets sent to me (Bob) and not the original sender (Alice)

Expected behaviour:

4. I hit reply and the From: field gets populated with bob@mydomain.com and the To: field with alice@mydomain.com (whilst both identities are listed in my user config it is bob@mydomain.com that is set to be the default and, from my perspective at least, the primary/preferred one where multiple valid options might exist)
5. The reply gets sent to the original sender (Alice)

Hope that makes sense?

**Comment by mjnewton** [comment]
I am struggling to reproduce this by sending myself an email from different identities. It does however reproduce consistently against a 'real' email (sent from Alice to Bob in my example).

**Comment by alecpl** [comment]
I suggest to test with a recent version like 1.6-rc.

As for the headers, the relevant ones are To, Cc, From, and Return-Path. There's also `identity_select` plugin that can use other headers for this purpose.

**Comment by mjnewton** [comment]
The From and Return-path headers both contain the sender (Alice) and the To the recipient (Bob) as you'd expect so nothing seemingly untoward there. No Cc header and the identity_select plugin is not enabled.

I have a planned update soon so will report on the outcome either way after that.

Thanks @alecpl.

**Comment by biredel** [comment]
I use Roundcube with multiple identities with this change, entirely disabling the offending code path:
```diff
diff --git a/program/include/rcmail_sendmail.php b/program/include/rcmail_sendmail.php
index 209bcbd46..ce0fb4ad0 100644
--- a/program/include/rcmail_sendmail.php
+++ b/program/include/rcmail_sendmail.php
@@ -1627,7 +1627,7 @@ class rcmail_sendmail
 
         $from_idx   = null;
         $found_idx  = ['to' => null, 'from' => null];
-        $check_from = in_array($mode, [self::MODE_DRAFT, self::MODE_EDIT, self::MODE_REPLY]);
+        $check_from = in_array($mode, [self::MODE_DRAFT, self::MODE_EDIT]);
 
         // Select identity
         foreach ($identities as $idx => $ident) {
```
What is the use case for swapping known identities anyway? Maybe if the conditions under which it produces better defaults were more clearly described, we could devise a better condition on when to enable it in the default installation.

**Comment by mjnewton** [comment]
Thanks @biredel - that's fixed it for me (in v1.4.11, making adjustments for the bracketing differences between the two sets of code).
